### ISSUE TITLE
Polling Trigger Does Not Produce Correct Metadata With "Emit All"

### ISSUE THREAD

**Issue by jhorbulyk** [elasticio/salesforce-component-v2#37]
# Component Bug Report

## Description
Dynamic metadata does not allign with what is produced for the polling trigger when the emit behavior is Select All 

## Component Version
* Version of the component running when the below steps to reproduce were performed: 2.2.1
* Platform version running when/installation name where the below steps were reproduced: 21.49
* Credentials used: Eio PS Account

## Steps to Reproduce

1. Create a flow with a single trigger: Salesforce V2.GetNewAndUpdatedObjectsPolling
2. On the input screen, select `Lead` for the *Object* field and `Emit all` for the *Output method*.
3. On the sample screen, click **Retrieve new sample from Salesforce v2**. Observe the JSON object produced.
4. On the sample screen, go back and click **Add sample manually** and then **Generate Stub Sample**. Observe the JSON object produced.

## Actual Result
* Generating a sample from the component results in JSON that looks like: `{"records":[{ .... Some Object ...}, { .... Some Other Object ...}, ....]}` 
* Generating a sample from the generate stub sample button results in `{"records":[{"Website":"minim"},{"ConvertedDate":"incididunt","ConvertedAccountId":"ex"}]}`
* Generating a sample from the generate stub sample button when *Output method* is `Emit all` looks nothing like the sample that is produced when *Output method* is `Emit individually`.

## Expected Result
* Clicking the **Generate Stub Sample** button when *Output method* is `Emit all` results in a sample that 
    * looks like `{"records":[{ .... Some Object ...}, { .... Some Other Object ...}, ....]}`
    * has each object in the array being as rich as the sample that is generated with *Output method* is `Emit individually`.

## Workaround(s)
Use the metadata provided when *Output method* is `Emit individually`.

**Comment by A3a3e1** [comment]
At a glance looks like a bug (or a feature ?) of a tool that is used behind the `Generate Stub Sample` button. 
A sample generated for a single object is full while an object based on the same json schema wrapped into an array is not.
The platform team needs time (a task) to investigate, they are not able to answer immediately.

**Comment by A3a3e1** [comment]
Under the hood this lib is being used for generating samples: https://www.npmjs.com/package/json-schema-faker
The platform currently uses version 0.4.1 while the last one is 0.5.0-rcv.40.
There is an option `alwaysFakeOptionals ` which when set to `true` solves the problem https://github.com/json-schema-faker/json-schema-faker/tree/0bff5c1f8aa1e314d348c23010bf078c4eb885eb/docs#available-options

There are 2 possible solutions:
1. Add `alwaysFakeOptionals: true` to the platform stub generation on the paltform side
2. Add `required: true` to every field in the metadata in the comnponent code. An ugly workaround but may help if the task is urgent

**Comment by jhorbulyk** [comment]
I've followed up with the customer that reported this and confirmed that the metadata being produced by the component is correct even if the generate sample button behaves strange artefacts. As a result, there is nothing that needs to be changed in the component code. 

There is no need to add required: true to every field in the metadata in the comnponent code. 

I've created https://eio.productboard.com/feature-board/775578-feature-organization/features/11317396/detail to cover potential improvements to the sample generation logic in the platform.
